pkg/util/version/version.go

```diff
@@ -234,9 +237,48 @@ func (v *Version) compareInternal(other *Version) int {
 		}
 	}
 
+	switch {
+	case len(oPR) < len(vPR):
+		return 1
+	case len(oPR) > len(vPR):
+		return -1
+	}
+
 	return 0
 }
 
+// Basic comparison functions
```

Comment:
This is unrelated to the bugfix (and I hadn't previously included functions like these because they weren't needed anywhere).

Reply:
I've got to the state where in kubeadm upgrade parts where a lot of version comparisons are happening (including selection of ranges) it would be good to have all basic comparison operators.

Reply:
OK. It should at least go in a separate commit (not one labeled "Fix version comparison for versions with preRelease components") and maybe should wait until the other PR where it's needed.